packages/react/src/components/SidePanel/index.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import Heading from '../Heading';
+import IconButton from '../IconButton';
+import Overlay from '../Overlay';
+
+const SidePanel = ({
+  children,
+  className,
+  footer,
+  isOpen: isOpenProp,
+  onClose,
+  title,
+  ...props
+}) => {
+  const [isOpen, setIsOpen] = React.useState(isOpenProp);
+
+  const toggleIsOpen = () => setIsOpen(!isOpen);
+
+  const handleClose = () => {
+    toggleIsOpen();
+    onClose();
+  };
+
+  React.useEffect(() => {
+    setIsOpen(isOpenProp);
+  }, [isOpenProp]);
+
+  return (
+    <>
+      <Overlay
+        className={cx({
+          'is-hidden': !isOpen,
+        })}
+        onClick={handleClose}
+      />
+
+      <div
+        className={cx('f-SidePanel', className, {
+          'is-open': isOpen,
+        })}
+        {...props}
+      >
+        <div className='f-SidePanel-header'>
+          <Heading level={4} as='h1'>
```

Comment:
Ok, I'm getting confused by our text classes again.
Heading level 4 - I read as `h4`
But that's followed by `h1`.
Either it's a mistake in this line or our `Heading` class is messed up and we need to fix it

Reply:
`as` is what component should be rendered
`level` is for styling purposes.

Since we don't want to style a component by its tag, we are relying on the `level` prop

By default `<Heading level={4} />` will render a `h4` with styles of a `h4`
But if you want this component to be a `h1` while keeping the same styles, you'll use the `as` prop

Reply:
👍